### PR TITLE
Migrating WT3 buffer tank unit

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:
Add buffer tank model

## Changes proposed in this PR:
- add unit model for buffer tank
- add corresponding test for buffer tank model

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/data/techno_economic/buffer_tank.yaml [hunk 1]
```diff
@@ -28,3 +28,8 @@ default:
   default_removal_frac_mass_solute:
     value: 0
     units: dimensionless
+  removal_frac_mass_solute:
```
Comment (adam-a-a) [inline watertap/data/techno_economic/buffer_tank.yaml]: this model is just a tank without any removal, so you should remove this

File: watertap/unit_models/zero_order/buffer_tank_zo.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    """
+
+    CONFIG = ZeroOrderBaseData.CONFIG()
+
+    def build(self):
+        super().build()
+
+        self._tech_type = "buffer_tank"
+
+        build_sido(self)
```
Comment (adam-a-a) [inline watertap/unit_models/zero_order/buffer_tank_zo.py]: I think this should be a pass-through model type, not SIDO.

**Review by adam-a-a — CHANGES_REQUESTED** [review]
This should be a pass-through unit instead of a SIDO. It doesn't remove or recover anything. Later, when we add costing, I suppose more parameters would be included to estimate costs of the tank, but that might be about it.

File: watertap/unit_models/zero_order/buffer_tank_zo.py (hunk 2 above)
Comment (luohezhiming) [reply watertap/unit_models/zero_order/buffer_tank_zo.py]: OK. Will modify this.

File: watertap/data/techno_economic/buffer_tank.yaml (hunk 1 above)
Comment (luohezhiming) [reply watertap/data/techno_economic/buffer_tank.yaml]: Removed.

**Review by bknueven — DISMISSED** [review]

**Review by adam-a-a — DISMISSED** [review]

**Comment by bknueven** [comment]
@luohezhiming this has some conflicts. If you like, I can try to resolve them.

**Comment by luohezhiming** [comment]
@bknueven I think I resolved those merge conflicts right now, let's see if all checks will pass.

**Review by bknueven — DISMISSED** [review]

**Review by adam-a-a — DISMISSED** [review]

**Comment by bknueven** [comment]
Closing because `git` is confused by our similar-looking models. Replaced with #377.

**Pull request closed without merging**